packages/web-pkg/src/composables/actions/spaces/useSpaceActionsNavigateToTrash.ts

```diff
@@ -0,0 +1,51 @@
+import { computed } from 'vue'
+import { SpaceAction } from '../types'
+import { useGettext } from 'vue3-gettext'
+import { useRouter } from '../../router'
+import { SpaceResource } from '@ownclouders/web-client'
+import { createLocationTrash } from '../../../router'
+import { createFileRouteOptions } from '../../../helpers'
+import { isProjectSpaceResource } from '@ownclouders/web-client/src/helpers'
+
+export const useSpaceActionsNavigateToTrash = () => {
+  const router = useRouter()
+  const { $gettext } = useGettext()
+
+  const getTrashLink = (space: SpaceResource) => {
+    return createLocationTrash('files-trash-generic', {
+      ...createFileRouteOptions(space, { fileId: space.fileId })
+    })
+  }
+
+  const actions = computed((): SpaceAction[] => [
+    {
+      name: 'navigateToTrash',
+      icon: 'delete-bin',
+      label: () => $gettext('Show deleted files'),
+      handler: ({ resources }) => {
+        return router.push(getTrashLink(resources[0])) as Promise<void>
```

Comment:
Could you remove `as Promise<void>` and not return the router push result? Feels a bit hacky to assert a `void` return if the handler actually returns something.